### PR TITLE
feat: Make docs tasks fail on warnings.

### PR DESCRIPTION
**Pull request by feanil**
Sphinx does a lot of useful error messaging as warnings.  Update the
default tox target in the cookiecutter to fail on warnings so that the
warnings can get caught and fixed.

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

### CODE REVIEW

File: cookiecutter-django-app/{{cookiecutter.repo_name}}/tox.ini [hunk 1]
```diff
@@ -55,8 +56,8 @@ commands =
     doc8 --ignore-path docs/_build README.rst docs
     rm -f docs/{{ cookiecutter.app_name }}.rst
     rm -f docs/modules.rst
-    make -C docs clean
-    make -C docs html
+    make -e -C docs clean
+    make -e -C docs html
```
Comment (timmc-edx) [inline cookiecutter-django-app/{{cookiecutter.repo_name}}/tox.ini:60]: Any reason not to just put `SPHINXOPTS` into the Makefile?

Comment (nedbat) [reply cookiecutter-django-app/{{cookiecutter.repo_name}}/tox.ini:60]: I was wondering that also.

File: python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py [hunk 2]
```diff
@@ -207,7 +207,7 @@ def get_version(*file_paths):
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ['_static']
+# html_static_path = ['_static']
```
Comment (timmc-edx) [inline python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py:210]: Do these other changes address some kind of warning?

Comment (feanil) [reply python-template/{{cookiecutter.placeholder_repo_name}}/docs/conf.py:210]: This directory didn't exist but was included which was resulting in an error.

File: cookiecutter-django-app/{{cookiecutter.repo_name}}/tox.ini (hunk 1 above)
Comment (feanil) [reply cookiecutter-django-app/{{cookiecutter.repo_name}}/tox.ini:60]: The Makefiles were initially generated so I chose to leave them alone and opted for putting the overrides in our tox file.

Comment (timmc-edx) [reply cookiecutter-django-app/{{cookiecutter.repo_name}}/tox.ini:60]: Ah, I see. Probably worth a comment here.

**Review by timmc-edx — APPROVED** [review]

**Pull request merged**